cnd/src/http_api/routes/index.rs

```diff
@@ -89,7 +89,19 @@ pub async fn post_han_ethereum_halight_bitcoin(
 
     facade.save(id, ()).await;
 
-    facade.initiate_communication(id, body.into()).await;
+    // This is kinda dirty, we should properly restrict the DialInformation by role
```

Comment:
A better way of solving this would be to move the role to the URL, based on that, you can have dedicated structs of each side and let serde handle this error :)

Reply:
That is a bigger refactoring though because it needs changes to the SDK, so I wouldn't do it right now!

cc @bonomat @D4nte : Some nice validation here that we should iterate on our stuff in the e2e tests before pushing it to the SDK!